mllib/src/main/scala/org/apache/spark/mllib/linalg/distributed/RowMatrix.scala

```diff
@@ -117,6 +117,7 @@ class RowMatrix @Since("1.0.0") (
     // Computes n*(n+1)/2, avoiding overflow in the multiplication.
     // This succeeds when n <= 65535, which is checked above
     val nt = if (n % 2 == 0) ((n / 2) * (n + 1)) else (n * ((n + 1) / 2))
+    val gramianSizeInMb = (nt / 1000000) * 8
```

Comment:
I think this needs to be `(nt * 8) / 1000000` to be correct, but `nt` could overflow, so needs to be `(nt * 8L) / 1000000` for example. You can cast back to int then.

Reply:
I'm not sure why, is it because (nt  / 1M) * 8 would  be hurt by double precision issues after nt/1M ?

Anyway I'll apply your suggestion

Reply:
These are integers though. (999999/1000000)*8 == 0, but (999999*8)/1000000 == 7, which is more like correct. It has to be done using longs though to avoid overflow: `((nt * 8L) / 1000000).toInt`